src/Mealz/MealBundle/Entity/ParticipantRepository.php

```diff
@@ -188,6 +184,55 @@ public function groupParticipantsByName($participations)
         return $result;
     }
 
+    /**
+     * @psalm-return array<string, array<string, list<Participant>>>
+     */
+    public function findAllGroupedBySlotAndProfileID(DateTime $date): array
+    {
+        $queryBuilder = $this->createQueryBuilder('p');
+        $queryBuilder
+            ->select(['p', 'm', 's', 'up'])
+            ->addSelect('COALESCE(IDENTITY(p.slot), 0) AS HIDDEN null_sort_col') // Pseudo col. to mark participants w/o slot
+            ->leftJoin('p.slot', 's')
+            ->join('p.profile', 'up')
+            ->join('p.meal', 'm', Join::WITH, 'm.dateTime >= :startTime AND m.dateTime <= :endTime')
+            ->orderBy('null_sort_col', 'DESC')
+            ->addOrderBy('s.order', 'ASC')
+            ->setParameters([
+                'startTime' => (clone $date)->setTime(0, 0),
+                'endTime' =>  (clone $date)->setTime(23, 59),
+            ]);
+
+        $result = $queryBuilder->getQuery()->getResult();
+        if (!is_array($result) || empty($result)) {
+            return [];
+        }
+
+        return $this->groupBySlotAndProfileID($result);
+    }
+
+    /**
+     * @param Participant[] $participants
+     *
+     * @psalm-return array<string, array<string, list<Participant>>>
+     */
+    private function groupBySlotAndProfileID(array $participants): array
+    {
+        $groupedParticipants = [];
+
+        foreach ($participants as $participant) {
+            $slot = $participant->getSlot();
+            if (null === $slot || $slot->isDisabled()) {
+                $groupedParticipants[''][$participant->getProfile()->getUsername()][] = $participant;
```

Comment:
Empty string Array-Key ist nicht so schön. Wofür ist es gedacht? Musst du die Daten im Array ablegen? Ergbit vielleicht ein Key "no-slot", "disabled-slot" oder sowas in der Richtung Sinn?